### ISSUE TITLE
Focus on the rows in the grid auto-selects the row instead of user pressing space key

### ISSUE THREAD

**Issue by sandeepnmenon** [OfficeDev/office-ui-fabric-react#10491]
### Environment Information

- **Package version(s)**: (6.182.1)
- **Browser and OS versions**: (Edge/Chrome and Windows 10)

> Note that our screen reader support is limited to Edge + Narrator. Issues that do not reproduce in Edge + Narrator, and aren't caused by obvious invalid aria values, should be filed with the respective screen reading software, not the Fabric repo.

### Issue:

When focus comes to any of the rows in the grid, the row is automatically getting selected without any user action.
Also, the selection made in a row is cleared when navigating to other rows via keyboard. This does not allow keyboard users to select multiple rows that are not adjacent to each other

Related issue: On selecting all rows using the top left select all. Then on navigating down using keyboard clears the selecting. This does not allow keyboard users t to select all and then unselect a few items.

### Reproduction of the issue in a codepen:
https://codepen.io/sandeepnmenon/pen/NWKOgYo?&editable=true

#### Actual behavior:
1. When focus comes to any of the rows in the grid, the row is automatically getting selected without any user action
2. User is unable to select multiple rows that are not adjacent using keyboard as the selection is getting cleared on navigation via keyboard

#### Expected behavior:

1. When focus comes to any of the rows in the grid, respective row should be selected only after user press space key
2. User should be able to select multiple rows that are not adjacent using keyboard

**Comment by aneeshack4** [comment]
@sandeepnmenon What is "respective is row"? What are the exact steps to repro this? 

And I believe the other behaviors are intended by design - only allowing selection of adjacent rows and clearing the selection of all rows after navigating into the rows. I'll check with the component owners & report back about this.

**Comment by aneeshack4** [comment]
@sandeepnmenon Please hold "CTRL" (or "CMD" on Mac) to navigate to non-adjacent rows and toggle selection with "SPACE" - this can be used to keyboard down to the rows without clearing all selected rows too.

**Comment by sandeepnmenon** [comment]
@aneeshack4  Thank you for the response.
Holding "CTRL" (or "CMD" on Mac) to navigate solves the second issue mentioned.

The first issue:
When focus comes to any of the rows in the grid via tabbing or via arrow keys,the row is selected automatically. (Without the user having to press space key)
Please close the issue, if this behavior is by design

**Comment by aneeshack4** [comment]
@sandeepnmenon Have you tried setting `disableAutoSelectOnInputElements` to true? https://developer.microsoft.com/en-us/fabric#/controls/web/references/selection#ISelectionZoneProps

**Comment by sandeepnmenon** [comment]
@aneeshack4  Thank you for the comment. Although `disableAutoSelectOnInputElements  `did not work,
`isSelectedOnFocus `to false did work.

However, a specific scenario is failing.

Repro:
1. Tab till select all button
2. Press down arrow to see the first item being selected automatically. (Further pressing down arrow does not auto-select any more elements.)

Issue reproduced in below codepen
https://codepen.io/sandeepnmenon/pen/NWKOgYo?&editable=true

PS: Pressing tab from select all button does not auto-select

**Comment by aneeshack4** [comment]
@sandeepnmenon Glad `isSelectedOnFocus` fixed it for you! Going to close this issue since it's resolved. This new scenario you mentioned will be fixed by a current PR that's out #10548.
